### PR TITLE
Refs #34118 -- Used Python 3.12 for GitHub actions.

### PR DESCRIPTION
**Pull request by felixxm**
[​https://www.python.org/downloads/release/python-3120/](https://www.python.org/downloads/release/python-3120/)

### CODE REVIEW

File: tests/requirements/py3.txt [hunk 1]
```diff
@@ -4,10 +4,10 @@ argon2-cffi >= 19.2.0
 bcrypt
 black
 docutils >= 0.19
-geoip2; python_version < '3.12'
+geoip2
 jinja2 >= 2.11.0
-numpy; python_version < '3.12'
-Pillow >= 6.2.1; sys.platform != 'win32' or python_version < '3.12'
+numpy
+Pillow >= 6.2.1; sys.platform != 'win32'
```
Comment (chicheng) [inline tests/requirements/py3.txt:10]: `sys.platform != 'win32'` should be removed as well?

Previously it will install on Windows if Python version is < 3.12 and now it won't install on Windows no matter the python version.

Comment (felixxm) [reply tests/requirements/py3.txt:10]: True, updated. Thanks :+1:

**Comment by cheng-liven** [comment]
[/.github/workflows/tests.yml#52](https://github.com/django/django/blob/main/.github/workflows/tests.yml#L52) and [/.github/workflows/schedule_tests.yml#48](https://github.com/django/django/blob/main/.github/workflows/schedule_tests.yml#L48) have outdated `node-version:` (14 and 16 used, both EOLed)

**Comment by felixxm** [comment]
> [/.github/workflows/tests.yml#52](https://github.com/django/django/blob/main/.github/workflows/tests.yml#L52) and [/.github/workflows/schedule_tests.yml#48](https://github.com/django/django/blob/main/.github/workflows/schedule_tests.yml#L48) have outdated `node-version:` (14 and 16 used, both EOLed)

This is unrelated to bumping Python version and should be handled separately.

**Comment by felixxm** [comment]
Waiting for a new ` aiohttp` release.

File: docs/intro/reusable-apps.txt [hunk 2]
```diff
@@ -240,10 +240,10 @@ this. For a small app like polls, this process isn't too difficult.
 
 #. Only Python modules and packages are included in the package by default. To
    include additional files, we'll need to create a ``MANIFEST.in`` file. The
-   setuptools docs referred to in the previous step discuss this file in more
-   detail. To include the templates, the ``README.rst`` and our ``LICENSE``
-   file, create a file ``django-polls/MANIFEST.in`` with the following
-   contents:
+   ``setuptools`` docs referred to in the previous step discuss this file in
```
Comment (smithdc1) [inline docs/intro/reusable-apps.txt:243]: The `setuptools` changes look unrelated to using 3.12 in the workflows? 🤔

Comment (felixxm) [reply docs/intro/reusable-apps.txt:243]: Without this, spelling check will fail when Python 3.12 is used. I can move it to the separate PR.

Comment (felixxm) [reply docs/intro/reusable-apps.txt:243]: Check out #17348.

**Comment by felixxm** [comment]
`aiohttp` 3.9.0 has been released :tada: We can finally use Python 3.12.

**Review by sarahboyce — APPROVED** [review]

**Comment by felixxm** [comment]
@sarahboyce Thanks for checking :+1:
